### ISSUE TITLE
AssignedByValParameterQuickFix leaks out of the scope of the procedure in properties.

### ISSUE THREAD

**Issue by comintern** [rubberduck-vba/Rubberduck#2660]
Just documenting so I don't forget to fix this.  Given the following code...
```
Option Explicit

Public Property Let Foo(ByVal bar As Long)
    bar = 42
End Property

Public Property Get Foo() As Long
    Foo = bar
End Property

Public Function bar() As Long
    bar = 42
End Function
```
...applying the quick fix to introduce a local parameter gives the following result:
```
Option Explicit

Public Property Let Foo(ByVal bar As Long)
Dim localBar As Long
localBar = bar
    localBar = 42
End Property

Public Property Get Foo() As Long
    Foo = localBar
End Property

Public Function localBar() As Long
    localBar = 42
End Function
```

**Comment by BZngr** [comment]
This is a bug I would like to correct as it was my implementation of the quickfix that introduced it....That is, unless you already have worked on a fix and have it ready...Let me know. Thanks.

**Comment by comintern** [comment]
@BZngr - Absolutely (thanks for the PR BTW).  There are a couple approaches for finding the last line of a procedure:

You can use the parser context attached to the `Declaration` and walk up the tree.  `Context.Parent` should always be a `VBAParser.ArgListContext`, and the parent of _that_ should be the procedure context:

![untitled](https://cloud.githubusercontent.com/assets/8944005/22739193/3bafc558-edd0-11e6-9c9b-8a02358a6092.png)

You could also call `CodeModule.GetProcCountLines()` and `module.GetProcBodyStartLine()` and calculate it (or check `CodeModule.GetProcOfLine()` instead of testing for the `End Sub` text).  Any of these should also account for line continuations:
```
Sub Foo(ByVal Bar As Long)
    Bar = 42
End _
Sub
```

In this specific instance though, you can just use the `Declaration.References` property of the inspection target to get all the usages and just pull and replace each line of code that contains a reference individually.  That approach is safe for this quick-fix because the `IdentifierReference.Context` is guaranteed to only span a single line (it can only be a single token). 

Make sure to synch your repo first - I ran this through ReSharper and made it localizable with #2661 (not merged yet, but you can synch it from [my fork](https://github.com/comintern/Rubberduck) if it isn't in `next` when you get started).

**Comment by BZngr** [comment]
@comintern - thanks for this is great information  Knew there had to be a better way.

**Comment by BZngr** [comment]
This is a bug I would like to correct as it was my implementation of the quickfix that introduced it....That is, unless you already have worked on a fix and have it ready...Let me know. Thanks.

**Comment by BZngr** [comment]
@comintern - I'm very close to finishing the correction for this bug.  While working with the CodeModule, I've observed a difference in the results obtained from the "starting line" procedures.  For the AssignedByValParameter feature, I need to have access to all the tokens used in the Procedure-under-repair so that I can detect when the user has specified a variable name that will collide with an existing name in the procedure.  A fragment from my current code looks like:

            _var module = Selection.QualifiedName.Component.CodeModule;
            var procName = module.GetProcOfLine(Selection.Selection.StartLine);
            var procKind = module.GetProcKindOfLine(Selection.Selection.StartLine);
            var numLines = module.GetProcCountLines(procName, procKind);_

            _//the below 'startLine' functions result in different answers - using the Select.Section.StartLine result
            var startLine = module.GetProcStartLine(procName, procKind);
            var selectionStartLine = Selection.Selection.StartLine;_

Since the StartLine values are different, calling CodeModule.GetLines(startline, numLines) results in a different set of results.  (BTW: GetProcCountLines(...) also seems to return a number larger than expected - but rather than assume/consider it is a bug, I am starting to wonder if I am using these functions correctly.

Running my code against a simple VBA module containing only the code and comments below:

_'CodeModule -> GetProcCountLines(procName,procKind)
'       retrieves commented code lines as well

Private Sub ProcLinesPickupCommentedCode(ByVal barByVal As Long)
    barByVal = 42
End Sub

'Expected Result
'Private Sub ProcLinesPickupCommentedCode(ByVal bar As Long)
'Dim xBarByVal As Long
'xBarByVal = bar
'    xBarByVal = 42
'End Sub_

CodeModule GetProcStartLine(...) = 1
Selection.Selection.Startline = 5  <- seems 'more' correct
GetProcCountLines(...) = 17  <- expected 3

Do these results make sense to you? - And, does the code fragment from my code look like I am using these function calls appropriately?  Thanks in advance.

**Comment by Vogel612** [comment]
merged in #2678
